components/gitpod-cli/cmd/rebuild.go

```diff
@@ -162,47 +165,44 @@ func runRebuild(ctx context.Context, supervisorClient *supervisor.SupervisorClie
 	ImageBuildDuration := time.Since(imageBuildStartTime).Milliseconds()
 	event.Set("ImageBuildDuration", ImageBuildDuration)
 
-	err = TerminateExistingContainer()
+	err = TerminateExistingContainer(ctx)
 	if err != nil {
 		event.Set("ErrorCode", utils.SystemErrorCode)
 		return err
 	}
 
-	messages := []string{
-		"\n\nYou are now connected to the container",
-		"You can inspect the container and make sure the necessary tools & libraries are installed.",
-		"When you are done, just type exit to return to your Gitpod workspace\n",
-	}
-
-	welcomeMessage := strings.Join(messages, "\n")
+	welcomeMessage := strings.Join([]string{
+		"\n\nYou are now connected to the container.",
+		"Check if all tools and libraries you need are properly installed.",
+		"When you are done, type \"exit\" to return to your Gitpod workspace.\n",
+	}, "\n")
 
-	dockerRunCmd := exec.Command(
+	dockerRunCmd := exec.CommandContext(ctx,
 		dockerPath,
 		"run",
 		"--rm",
+		"-e", fmt.Sprintf("SHELL=%s", os.Getenv("SHELL")),
+		"-v", "/workspace:/workspace",
 		"--label", "gp-rebuild=true",
-		"-it",
-		tag,
-		"bash",
+		"-it", tag,
+		"/bin/sh",
 		"-c",
-		fmt.Sprintf("echo '%s'; bash", welcomeMessage),
+		fmt.Sprintf("echo \"%s\"; cd \"%s\"; if [ -x \"$(command -v $SHELL)\" ]; then $SHELL; else /bin/sh; fi", welcomeMessage, wsInfo.CheckoutLocation),
```

Comment:
@felladrin most of time SHELL will be empty, but /bin/bash will be there, we should try both options and only like last resort fallback to sh

Reply:
Ok, updated!